src/domain/Filters/AddFilterDropdownMenu/AddFilterDropdownMenu.tsx

```diff
@@ -24,129 +24,174 @@ interface IAddFilterDropdownMenuProps {
-    setFilterValue(option.value)
-  }
-
-  function valueInputOption (): Array<{ label: string, value: string | number | boolean }> | undefined {
-    const selectedFilter = filters.find((filter: IAddFilterDropdownMenuFilter) => filter.fieldName === filterField)
+}: IAddFilterDropdownMenuProps<FilterValue>) {
+  return (
+    <StyledDropdownMenu
+      toggleComponent={toggleComponent}
+      componentContext={componentContext}
+    >
+      {({ closeMenu }: { closeMenu: (() => void) }) => (
+        <DropdownMenuContent<FilterValue>
+          closeMenu={closeMenu}
+          filterMessage={filterMessage}
+          filters={filters}
+          onFilterAdded={onFilterAdded}
+        />
+      )}
+    </StyledDropdownMenu>
+  )
+}
 
+function DropdownMenuContent<FilterValue = string | number> (
+  {
+    closeMenu,
+    filterMessage,
+    filters,
+    onFilterAdded
+  }: IAddFilterDropdownMenuContentProps<FilterValue>) {
+  const [selectedFilterName, setSelectedFilterName] = useState<string | null>(null)
+  const [filterValue, setFilterValue] = useState<FilterValue | null>(null)
+
+  const filtersOptions = filters.map((filter) => (
+    {
+      label: filter.label,
+      value: filter.fieldName
+    }))
+
+  const selectedFilter = filters.find((filter) => filter.fieldName === selectedFilterName)
+  const onChange = useCallback((value) => setFilterValue(value), [setFilterValue])
+
+  const filterValueComponent = useMemo(() => {
     if (selectedFilter) {
-      return selectedFilter.selectOptions
+      if (selectedFilter.customInputComponent) {
+        const CustomInputComponent = selectedFilter.customInputComponent
+        return (
+          <CustomInputComponent
+            value={filterValue}
+            onChange={setFilterValue}
+          />
+        )
+      }
+
+      if (selectedFilter.selectOptions) {
+        return (
+          <SelectInput
+            name='filterDropdownValueInput'
+            value={(filterValue ?? '') as string | number}
+            options={selectedFilter.selectOptions as unknown as ISelectOption[]}
+            placeholder='Select a value'
+            isClearable={false}
+            onChange={onChange}
+          />
+        )
+      }
     }
-  }
 
-  function applyFilter (closeMenu: () => void) {
-    return () => {
-      const selectedFilter = filters.find((filter: IAddFilterDropdownMenuFilter) => filter.fieldName === filterField)
+    return null
+  }, [selectedFilter, filterValue])
 
-      if (selectedFilter && filterValue) {
-        const selectedFilterOption = selectedFilter.selectOptions.find((option) => option.value === filterValue)
+  const handleFilterSelectChange = useCallback((option) => setSelectedFilterName(option?.value), [setSelectedFilterName])
 
+  const applyFilter = useCallback(
+     () => {
+      if (selectedFilter && filterValue) {
+        const selectedFilterOption = selectedFilter.selectOptions?.find((option) => option.value === filterValue)
```

Comment:
`selectOptions` doesn't exist for a custom component, so I'm not sure how this works